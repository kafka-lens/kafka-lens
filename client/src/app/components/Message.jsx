import React from 'react';
import '../css/Message.scss';

const Message = ({ message, offset }) => (
  <span>
    <div className="single-message">
      {message}
      <p className="offset-indicator">{offset}</p>
    </div>
  </span>
);

export default Message;
