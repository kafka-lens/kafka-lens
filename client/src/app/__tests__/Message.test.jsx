import React from 'react';
import { shallow } from 'enzyme';

import Message from '../components/Message';

describe('Message.jsx unit tests', () => {
  let wrapper;
  const props = {
    offset: '5',
    message: 'Test',
  };
  beforeAll(() => {
    wrapper = shallow(<Message {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render a div tag with inner text', () => {
    expect(wrapper.find('div').text()).toEqual('Test5');
  });

  it('Should render a p tag with inner text', () => {
    expect(wrapper.find('p').text()).toEqual('5');
  });
});
