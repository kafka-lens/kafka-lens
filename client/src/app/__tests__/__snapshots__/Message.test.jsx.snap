// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.jsx unit tests Should render without errors 1`] = `
<span>
  <div
    className="single-message"
  >
    Test
    <p
      className="offset-indicator"
    >
      5
    </p>
  </div>
</span>
`;
